src/test/java/teammates/sqlui/webapi/UpdateNotificationActionTest.java

```diff
@@ -0,0 +1,217 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
+import teammates.storage.sqlentity.Notification;
+import teammates.ui.output.NotificationData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+import teammates.ui.request.NotificationUpdateRequest;
+import teammates.ui.webapi.InvalidHttpParameterException;
+import teammates.ui.webapi.UpdateNotificationAction;
+
+/**
+ * SUT: {@link UpdateNotificationAction}.
+ */
+public class UpdateNotificationActionTest extends BaseActionTest<UpdateNotificationAction> {
+    NotificationAttributes typicalNotificationAttribute;
+    Notification testNotification;
+    Notification newNotification;
+    NotificationUpdateRequest notificationRequest;
+    String[] requestParams;
+
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return PUT;
+    }
+
+    @BeforeMethod
+    void setUp() {
+        typicalNotificationAttribute = typicalBundle.notifications.get("notification1");
```

Comment:
Can use [this](https://github.com/TEAMMATES/teammates/blob/8005a6a3c571c5bf8406f101b8de00a231d4f2b5/src/test/java/teammates/test/BaseTestCase.java#L133) method instead